exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -59,12 +59,13 @@
   private final ListVectorOutput listOutput;
   private final boolean extended = true;
   private final boolean readNumbersAsDouble;
+  private List<String> path = Lists.newArrayList();
 
   /**
    * Collection for tracking nullable fields during reading
    * and storing them for creating default typed vectors
    */
-  private final Set<String> nullableFields  = Sets.newLinkedHashSet();
+  private final Set<List<String>> nullableFields  = Sets.newLinkedHashSet();
```

Comment:
Do you mean the iterating on line 163 needs the order to be preserved? I doubt it. Could you please double check?